src/language-js/print/assignment.js

```diff
@@ -67,18 +67,26 @@ function printAssignmentRight(leftNode, rightNode, printedRight, options) {
     return indent([line, printedRight]);
   }
 
-  /**
-   * @param {Node} node
-   * @returns {Node}
-   */
-  function getUnaryArgument(node) {
-    if (node.type === "UnaryExpression") {
-      return getUnaryArgument(node.argument);
-    }
-    return node;
+  if (canBreakAssignmentRight(leftNode, rightNode, options)) {
+    return group(indent([line, printedRight]));
   }
 
-  const canBreak =
+  return [" ", printedRight];
+}
+
+/**
+ * @param {Node} node
+ * @returns {Node}
+ */
+function getUnaryArgument(node) {
+  if (node.type === "UnaryExpression") {
+    return getUnaryArgument(node.argument);
+  }
+  return node;
+}
+
+function canBreakAssignmentRight(leftNode, rightNode, options) {
```

Comment:
I mean rewrite in `if-else`, hard to read.WDYT?

Reply:
like this?

```js
function canBreakAssignmentRight(leftNode, rightNode, options) {
  if (isBinaryish(rightNode) && !shouldInlineLogicalExpression(rightNode)) {
    return true;
  }
  if (
    rightNode.type === "ConditionalExpression" &&
    isBinaryish(rightNode.test) &&
    !shouldInlineLogicalExpression(rightNode.test)
  ) {
    return true;
  }
  if (rightNode.type === "StringLiteralTypeAnnotation") {
    return true;
  }
  if (
    rightNode.type === "ClassExpression" &&
    isNonEmptyArray(rightNode.decorators)
  ) {
    return true;
  }
  if (
    (leftNode.type === "Identifier" ||
      isStringLiteral(leftNode) ||
      leftNode.type === "MemberExpression") &&
    (isStringLiteral(getUnaryArgument(rightNode)) ||
      isMemberExpressionChain(getUnaryArgument(rightNode))) &&
    // do not put values on a separate line from the key in json
    options.parser !== "json" &&
    options.parser !== "json5"
  ) {
    return true;
  }
  if (rightNode.type === "SequenceExpression") {
    return true;
  }
  return false;
}
```

Reply:
More blank lines.

Reply:
And less ternary, still a lot.

Reply:
Hmm, I have no idea to less logical operators cleanly...